core/trino-main/src/test/java/io/trino/sql/planner/optimizations/TestAddLocalExchangesForTaskScaleWriters.java

```diff
@@ -326,4 +388,42 @@ public void testLocalScaledPartitionedWriterForConnectorPartitioning()
                                         exchange(REMOTE, REPARTITION, partitioningHandle,
                                                 tableScan("source_table", ImmutableMap.of("customer", "customer", "year", "year")))))));
     }
+
+    @Test
+    public void testLocalScaledPartitionedWriterWithoutPreferredPartitioning()
+    {
+        assertDistributedPlan(
+                "INSERT INTO system_partitioned_table SELECT * FROM source_table",
+                testSessionBuilder()
+                        .setCatalog("mock_report_written_bytes_with_multiple_writer_per_partition")
+                        .setSchema("mock")
+                        .setSystemProperty(TASK_SCALE_WRITERS_ENABLED, "true")
+                        .setSystemProperty(SCALE_WRITERS, "false")
+                        .build(),
+                anyTree(
+                        tableWriter(
+                                ImmutableList.of("customer", "year"),
+                                ImmutableList.of("customer", "year"),
+                                project(
+                                        exchange(LOCAL, REPARTITION, SCALED_WRITER_HASH_DISTRIBUTION,
+                                                exchange(REMOTE, REPARTITION, FIXED_ARBITRARY_DISTRIBUTION,
```

Comment:
Is it possible to have a test where CBO doesn't choose global partitioning (e.g. due to skewness), but we still enforce local one?